File: rt-master/src/main/java/xyz/redtorch/node/master/service/UserService.java
package xyz.redtorch.node.master.service;

import xyz.redtorch.node.master.po.UserPo;

import java.util.List;

public interface UserService {
    UserPo auth(UserPo user);

    String changePassword(UserPo user);

    List<UserPo> getUserList();

    void deleteUserByUsername(String username);

    void updateUserDescriptionByUsername(UserPo user);

    void updateUserPasswordByUsername(UserPo user);

    void addUser(UserPo user);

    void updateUserPermissionByUsername(UserPo user);

    UserPo getUserByUsername(String username);
}

File: rt-master/src/main/kotlin/xyz/redtorch/master/service/UserService.kt
package xyz.redtorch.master.service

import xyz.redtorch.common.storage.po.User

interface UserService {
    fun getUserList(): List<User>

    fun deleteUserById(userId: String)

    fun resetUserTokenById(userId: String): User?

    fun addUser(userId: String, description: String): User?

    fun userAuth(userId: String, token: String): User?

    fun updateUserDescriptionById(userId: String, description: String)

    fun upsertUserById(user: User)

    fun enableUserById(userId: String)

    fun banUserById(userId: String)

}